src/ConformalTracking.cc

```diff
@@ -1659,14 +1648,335 @@ void ConformalTracking::extendHighPT(UniqueKDTracks& conformalTracks, SharedKDCl
+    bool             firstLayer     = true;
+    bool             skipLayer      = false;
+    SKDCluster       expectedHit    = nullptr;
+
+    // Start the extension into input collection, layer by layer
+    do {
+      streamlog_out(DEBUG9) << "- Start extension" << std::endl;
+      // Find nearest neighbours in theta and sort them by layer
+      SKDCluster const& kdhit = seedCell->getEnd();
+      streamlog_out(DEBUG9) << "- Endpoint seed cell: [x,y] = [" << kdhit->getX() << ", " << kdhit->getY() << "]; "
+                            << "]; r = " << kdhit->getR() << "; [subdet,layer] = [" << kdhit->getSubdetector() << ", "
+                            << kdhit->getLayer() << "]" << std::endl;
+      double theta = kdhit->getTheta();
+      nearestNeighbours->allNeighboursInTheta(
+          theta, m_thetaRange * 4, results, [&kdhit, vertexToTracker](SKDCluster const& nhit) {
+            //if same subdet, take only the hits within two layers
+            if (nhit->getSubdetector() == kdhit->getSubdetector()) {
+              if ((vertexToTracker && nhit->getLayer() > (kdhit->getLayer() + 2)) ||
+                  (!vertexToTracker && nhit->getLayer() < (kdhit->getLayer() - 2))) {
+                return true;
+              }
+            }
+
+            if ((vertexToTracker && nhit->getR() > kdhit->getR()) || (!vertexToTracker && nhit->getR() < kdhit->getR()))
+              return true;
+            return false;
+          });
+      //nearestNeighbours->allNeighboursInRadius(kdhit, parameters._maxDistance, results);
+      streamlog_out(DEBUG9) << "- Found " << results.size() << " neighbours. " << std::endl;
+      if (streamlog::out.write<DEBUG9>()) {
+        for (auto const& neighbour : results) {
+          streamlog_out(DEBUG9) << "-- Neighbour from allNeighboursInTheta : [x,y] = [" << neighbour->getX() << ", "
+                                << neighbour->getY() << "]; r = " << neighbour->getR() << "; [subdet,layer] = ["
+                                << neighbour->getSubdetector() << ", " << neighbour->getLayer() << "]" << std::endl;
+        }
+      }
+
+      if (results.size() == 0) {
+        loop = false;
+        continue;
+      }
+
+      std::sort(results.begin(), results.end(), (vertexToTracker ? sort_by_layer : sort_by_lower_layer));
+      // Get the final values of subdet and layer to stop the loop at the (vertexToTracker? outermost : innermost) layer with neighbours
+      final_subdet = results.back()->getSubdetector();
+      final_layer  = results.back()->getLayer();
+      streamlog_out(DEBUG9) << "- Final layer with neighbours: [subdet,layer] = [" << final_subdet << ", " << final_layer
+                            << "]" << std::endl;
+
+      // If no hit was found on a layer, we use expectedHit as current position
+      // However, the nearest neighbour search is always performed from the last real hit (kdhit)
+      SKDCluster hitOnCurrentLayer = skipLayer ? expectedHit : kdhit;
```

Comment:
Can't this be a reference `SKDCluster&` ?

Reply:
No, I tried and it does not compile.

Reply:
Ahh, it needs to be `const&` because kdhit is also `const&`